ha/setup/ha_setup.py

```diff
@@ -0,0 +1,366 @@
+                raise HaConfigError("Cluster config failed; pcs not installed")
+            # If cluster is not created; create a cluster.
+            elif(err.find("cluster is not currently running on this node") != -1):
+                try:
+                    Log.info(f"Creating cluster: {cluster_name} with node: {minion_name}")
+                    cluster_auth(cluster_user, cluster_secret, nodelist)
+                    cluster_create(cluster_name, nodelist)
+                    Log.info(f"Created cluster: {cluster_name} successfully")
+                    Log.info("Creating pacemaker resources")
+                    create_all_resources()
+                    Log.info("Created pacemaker resources successfully")
+                except Exception as e:
+                    Log.error(f"Cluster creation failed; destroying the cluster. Error: {e}")
+                    output = self._execute.run_cmd(const.PCS_CLUSTER_DESTROY, check_error=True)
+                    Log.info(f"Cluster destroyed. Output: {output}")
+                    raise HaConfigException("Cluster creation failed")
+            else:
+                pass # Nothing to do
+        else:
+            # Cluster exists already, check if it is a new node and add it to the existing cluster.
+             Log.info("The cluster exists already, check and add new node")
+        Log.info("config command is successful")
+
+class InitCmd(Cmd):
+    """
+    Init Setup Cmd
+    """
+    name = "init"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process init command.
+        """
+        pass # TBD
+
+class TestCmd(Cmd):
+    """
+    Test Cmd
+    """
+    name = "test"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process test command.
+        """
+        pass # TBD: Write code here to check that all resources have been configured.
+
+class UpgradeCmd(Cmd):
+    """
+    Upgrade Cmd
+    """
+    name = "upgrade"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process upgrade command.
+        """
+        pass # TBD
+
+class ResetCmd(Cmd):
+    """
+    Reset Cmd
+    """
+    name = "reset"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process reset command.
+        """
+        pass # TBD
+
+class CleanupCmd(Cmd):
+    """
+    Cleanup Cmd
+    """
+    name = "cleanup"
+
+    def __init__(self, args):
+        """
+        Init method.
+        """
+        super().__init__(args)
+
+    def process(self):
+        """
+        Process cleanup command.
+        """
+        Log.info("Processing cleanup command")
```

Comment:
A try..except block would be required here too and an HaCleanupException.

Reply:
done